src/main.rs

```diff
@@ -576,12 +587,14 @@ struct Stacked {
     lr: u32,
     pc: u32,
     xpsr: u32,
+    fpu_regs: Option<StackedFpuRegs>,
 }
 
 impl Stacked {
-    fn read(core: &mut Core<'_>, sp: u32) -> Result<Self, anyhow::Error> {
-        let mut registers = [0; 8];
-        core.read_32(sp, &mut registers)?;
+    fn read(core: &mut Core<'_>, sp: u32, fpu: bool) -> Result<Self, anyhow::Error> {
+        let mut storage = [0; 8 + 17];
```

Comment:
could we turn these "8" and "17" into constants? Perhaps as associated constants of `StackedFpuRegs`? Maybe even defined as `mem::size_of::<Self>()`?